include/ProgramRepresentation/ProgramFunction.h

```diff
@@ -12,28 +12,33 @@ class ProgramFunction {
   std::string functionName;
 
 public:
-  static void logoutPF(ProgramFunction &pf);
+  // debugging function that lists program points and methods called (methods
+  // logged if logMethods is true) of a program function
+  static void logoutProgramFunction(ProgramFunction &programFunction,
+                                    bool logMethods);
+
   ProgramFunction();
   ProgramFunction(std::string functionName);
 
   void addProgramPoint(ProgramPoint programPoint);
 
   void setProgramPoint(std::string pointName, ProgramPoint programPoint);
 
-  std::list<ProgramPoint> getProgramPoints();
+  std::list<ProgramPoint> getProgramPoints() const;
 
   // returns a program point based off pointName and, if addNewIfNotFound is
   // true, creates a new one if it was not found. if addNewIfNotFound is false
   // and the point was not found, the program fails & exits
-  ProgramPoint getProgramPoint(std::string pointName, bool addNewIfNotFound);
+  ProgramPoint getProgramPoint(const std::string &pointName,
+                               bool addNewIfNotFound);
 
   // returns a program point ref based off name and, if addNewIfNotFound is
   // true, creates a new one if it was not. if addNewIfNotFound is false and the
   // point was not found, the program fails & exits
-  ProgramPoint *getProgramPointRef(std::string pointName,
+  ProgramPoint *getProgramPointRef(const std::string &pointName,
```

Comment:
They're still correct since this change is really for documentation, making it clear that `pointName` has not changed. Additionally, the compiler would have yelled at me prior if they were wrong (and it has not!)